File: coeficienteangular.py
def calcular_coeficiente_angular():
    print("--- Calculadora de Coeficiente Angular ---")
    print("Insira as coordenadas do Ponto 1 (x1, y1):")

    try:
        x1 = float(input("x1: "))
        y1 = float(input("y1: "))
        
        print("\nInsira as coordenadas do Ponto 2 (x2, y2):")
        x2 = float(input("x2: "))
        y2 = float(input("y2: "))
        
        # Cálculo das variações (Deltas)
        delta_y = y2 - y1
        delta_x = x2 - x1

        if delta_x == 0:
            print("\nResultado: O coeficiente angular é INDEFINIDO.")
            print("Explicação: A reta é vertical (paralela ao eixo Y).")
        else:
            m = delta_y / delta_x
            print(f"\nResultado: O coeficiente angular (m) é: {m:.2f}")
            
            # Informação extra sobre a inclinação
            if m > 0:
                print("A reta é crescente.")
            elif m < 0:
                print("A reta é decrescente.")
            else:
                print("A reta é horizontal (constante).")

    except ValueError:
        print("\nErro: Por favor, insira apenas números válidos.")

    

File: coeficiente_angular.c
#include <stdio.h>

int main() {
    float x1, y1, x2, y2;
    float m;

    printf("--- Calculadora de Coeficiente Angular ---\n\n");

    // Entrada de dados do primeiro ponto
    printf("Digite as coordenadas do ponto A (x1 y1): ");
    scanf("%f %f", &x1, &y1);

    // Entrada de dados do segundo ponto
    printf("Digite as coordenadas do ponto B (x2 y2): ");
    scanf("%f %f", &x2, &y2);

    // Validação para evitar divisão por zero (reta vertical)
    if (x2 == x1) {
        printf("\nErro: x1 nao pode ser igual a x2.\n");
        printf("A reta e vertical e o coeficiente angular e indefinido.\n");
    } else {
        // Cálculo do coeficiente angular
        m = (y2 - y1) / (x2 - x1);
        
        printf("\n-----------------------------------------\n");
        printf("O coeficiente angular (m) da reta e: %.2f\n", m);
        printf("-----------------------------------------\n");
    }

    return 0;
}